### PR TITLE
schedule: refactor operator influence

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve? 
Currently, we have `OpInfluence` for each scheduler, but only a few of them will be used.

### What is changed and how it works?
This PR moves `OpInfluence` related logic from `basic_cluster.go` into `operator_controller.go`, simplifies `Schedule` interface, and provide `GetOpInfluence` and `SetOpInfluence` function to access `OpInfluence`.

### Check List 

Tests 

 - Manual test (`make test`)

### CODE REVIEW

File: server/schedule/operator_controller.go [hunk 1]
```diff
@@ -316,3 +318,100 @@ func (oc *OperatorController) OperatorCount(mask OperatorKind) uint64 {
 	}
 	return total
 }
+
+// OpInfluenceFilterOption used to filter opInfluence.
+type OpInfluenceFilterOption func(opInfluence OpInfluence)
+
+// GetOpInfluence gets OpInfluence.
+func (oc *OperatorController) GetOpInfluence(opts ...OpInfluenceFilterOption) OpInfluence {
```
Comment (nolouch) [inline server/schedule/operator_controller.go]: Maybe we can use `Cluster` as a parameter, and use the cluster to distinguish operators.

File: server/schedulers/scatter_range.go [hunk 2]
```diff
@@ -75,30 +74,18 @@ func (l *scatterRangeScheduler) IsScheduleAllowed(cluster schedule.Cluster) bool
 	return l.opController.OperatorCount(schedule.OpRange) < cluster.GetRegionScheduleLimit()
 }
 
-func (l *scatterRangeScheduler) getOperators(opInfuence schedule.OpInfluence) []*schedule.Operator {
```
Comment (disksing) [inline server/schedulers/scatter_range.go]: Is it correct to remove this? /cc @nolouch

Comment (disksing) [reply server/schedulers/scatter_range.go]: The rest LGTM. PTAL @nolouch

Comment (nolouch) [reply server/schedulers/scatter_range.go]: Yes, It is correct. `GetOpInfluence` handle it.

**Pull request merged**